File: app/src/main/java/com/expansion/lg/kimaru/expansion/other/WifiState.java
package com.expansion.lg.kimaru.expansion.other;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.util.Log;

import java.net.InetAddress;
import java.net.InetSocketAddress;
import java.net.Socket;
import java.net.SocketAddress;

/**
 * Created by kimaru on 4/13/17.
 */

public class WifiState {
    ConnectivityManager connectivityManager;
    NetworkInfo wifiCheck;
    Context context;

    public WifiState(Context context){
        this.context = context;
        this.connectivityManager = (ConnectivityManager)
                context.getSystemService(context.CONNECTIVITY_SERVICE);
        if (connectivityManager != null)
            this.wifiCheck = connectivityManager.getActiveNetworkInfo();
    }

    public boolean isWifiConnected(){
        boolean wifi;
        try {
            return wifiCheck.getType() == ConnectivityManager.TYPE_WIFI;
        }catch (Exception e){
            return false;
        }

    }

    public boolean canReachPeerServer(){
        Log.d("Tremap", "Checking Peer connection");
        return isWifiConnected();
    }

    public boolean canReachServer(InetAddress ip, int port){
        boolean canReach = false;
        try{
            SocketAddress socketAddress = new InetSocketAddress(ip, port);
            Socket socket = new Socket();
            int timeOutMs = 2000;
            socket.connect(socketAddress, timeOutMs);
            canReach = true;
        }catch (Exception e){

        }

        return canReach;
    }


}

File: app/src/main/java/com/expansion/lg/kimaru/expansion/other/WifiState.kt
package com.expansion.lg.kimaru.expansion.other

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import android.util.Log

import java.net.InetAddress
import java.net.InetSocketAddress
import java.net.Socket
import java.net.SocketAddress

/**
 * Created by kimaru on 4/13/17.
 */

class WifiState(internal var context: Context) {
    internal var connectivityManager: ConnectivityManager? = null
    internal var wifiCheck: NetworkInfo? = null

    val isWifiConnected: Boolean
        get() {
            val wifi: Boolean
            try {
                return wifiCheck!!.type == ConnectivityManager.TYPE_WIFI
            } catch (e: Exception) {
                return false
            }

        }

    init {
        this.connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        this.wifiCheck = if (connectivityManager != null) connectivityManager!!.activeNetworkInfo else null
    }

    fun canReachPeerServer(): Boolean {
        Log.d("Tremap", "Checking Peer connection")
        return isWifiConnected
    }

    fun canReachServer(ip: InetAddress, port: Int): Boolean {
        var canReach = false
        try {
            val socketAddress = InetSocketAddress(ip, port)
            val socket = Socket()
            val timeOutMs = 2000
            socket.connect(socketAddress, timeOutMs)
            canReach = true
        } catch (e: Exception) {

        }

        return canReach
    }


}
